packages/admin/src/form/FinalFormMultiSelect.tsx

```diff
@@ -0,0 +1,132 @@
+import { Check } from "@comet/admin-icons";
+import { createStyles, List, ListItem, ListItemIcon, ListItemText, WithStyles, withStyles } from "@material-ui/core";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+/*---------------- FinalFormMultiSelectListItemComponent ----------------*/
+
+type FinalFormMultiSelectListDefaultItemClassKey = "listItemIcon" | "listItemText";
+
+const listItemStyles = () => {
+    return createStyles<FinalFormMultiSelectListDefaultItemClassKey, FinalFormMultiSelectListDefaultItemProps>({
+        listItemIcon: {},
+        listItemText: {},
+    });
+};
+
+interface FinalFormMultiSelectListDefaultItemProps {
+    option: Option;
+    isSelected: boolean;
+}
+
+function FinalFormMultiSelectListDefaultItemComponent({
+    option,
+    isSelected,
+    classes,
+}: WithStyles<typeof listItemStyles> & FinalFormMultiSelectListDefaultItemProps) {
+    return (
+        <>
+            {option.icon && (
+                <ListItemIcon
+                    classes={{
+                        root: classes.listItemIcon,
+                    }}
+                >
+                    {option.icon}
+                </ListItemIcon>
+            )}
+            <ListItemText
+                classes={{
+                    root: classes.listItemText,
+                }}
+            >
+                {option.label}
+            </ListItemText>
+            {isSelected && <Check />}
+        </>
+    );
+}
+
+const FinalFormMultiSelectListDefaultItem = withStyles(listItemStyles, { name: "CometAdminFinalFormMultiSelectListDefaultItem" })(
+    FinalFormMultiSelectListDefaultItemComponent,
+);
+
+declare module "@material-ui/core/styles/overrides" {
+    interface ComponentNameToClassKey {
+        CometAdminFinalFormMultiSelectListDefaultItem: FinalFormMultiSelectListDefaultItemClassKey;
+    }
+}
+
+/*---------------- FinalFormMultiSelectComponent ----------------*/
+
+export interface Option {
+    label: string | React.ReactNode;
+    value: string;
+    icon?: React.ReactNode;
+}
+
+interface FinalFormMultiSelectProps extends FieldRenderProps<string, HTMLDivElement> {
+    options: Option[];
+    renderItem?: (option: Option) => React.ReactNode;
+}
+
+export type FinalFormMultiSelectClassKey = "listRoot" | "listItemRoot" | "listItemRootSelected";
```

Comment:
To stay consistent with other components, i think `listRoot` should just be `root`.  
Also I don't think the list-item should include `root`.

```suggestion
export type FinalFormMultiSelectClassKey = "root" | "listItem" | "listItemSelected";
```